lxd/storage/drivers/utils.go

```diff
@@ -139,3 +141,20 @@ func vfsResources(path string) (*api.ResourcesStoragePool, error) {
 
 	return &res, nil
 }
+
+// GetPoolMountPoint returns the mountpoint of the given pool.
+// {LXD_DIR}/storage-pools/<pool>
+func GetPoolMountPoint(poolName string) string {
+	return shared.VarPath("storage-pools", poolName)
+}
+
+// GetVolumeMountPoint returns the mount path for a specific volume based on its pool and type and
+// whether it is a snapshot or not.
+// For VolumeTypeImage the volName is the image fingerprint.
+func GetVolumeMountPoint(poolName string, volType VolumeType, volName string) string {
```

Comment:
@stgraber here is the unified GetVolumeMountPoint, with its detection of snapshots via volume name. I added some tests below too.